### PR TITLE
remove the dependence between tables and table

### PR DESCRIPTION
**Pull request by astaxie**
As tables is the implement of table. So tables like the drivers, while
table like database/sql. while drivers need to regiester in the init.

### CODE REVIEW

File: infoschema/infoschema.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/pingcap/tidb/meta/autoid"
 	"github.com/pingcap/tidb/model"
 	"github.com/pingcap/tidb/table"
+	_ "github.com/pingcap/tidb/table/tables"
```
Comment (c4pt0r) [inline infoschema/infoschema.go]: any elegant way to do this?

Comment (astaxie) [inline infoschema/infoschema.go]: It's the best way for implement and interface.

the same style as `database/sql`

```
import "database/sql"
import _ "github.com/go-sql-driver/mysql"

db, err := sql.Open("mysql", "user:password@/dbname")
```

Comment (coocood) [inline infoschema/infoschema.go]: golint reports 
"infoschema.go:25:2: a blank import should be only in a main or test package, or have a comment justifying it"

Comment (astaxie) [inline infoschema/infoschema.go]: @coocood good catch. https://github.com/astaxie/tidb/commit/bb78a3414b6271c1cbbe8aebcc74e675de4d6fb9

**Comment by ngaut** [comment]
LGTM

**Comment by c4pt0r** [comment]
LGTM

**Pull request merged**